### ISSUE TITLE
"Failed to unlink file" when moving files / merging directories, dataloss

### ISSUE THREAD

**Issue by aleksejrs** [vifm/vifm#391]
vifm (0.10-1) from https://packages.debian.org/buster/vifm
:version says Version: 0.10, Git info: built out of repository.
I updated Vifm from 0.9.1-1 on Nov 24.
Debian testing, Btrfs, used OverlayFS but not likely to have character devices everywhere.

* It happens also without OverlayFS.
* It happened on at least two different disks (one was OverlayFS' upper directory at the time, and the other was one where OverlayFS was not running at the time, and the lost directories were unlikely to have any).

I am not going to try right now (it's night), but I can remember that:
1. It happens most likely during directory merges containing files that have to be overwritten.
2. I agree to overwrite a file.
3. Vifm says at some point something like (this just happened without OverlayFS) "Failed to unlink file" or "directory", or (maybe it happened with OverlayFS today) "Can't read " `<the target (inode?) which should have resulted from the move>`.
4. The process of moving ends prematurely.

Morning. In case this ends up being related: In the moved directory tree, some directories belong to root `drwxr-x--- 1 root root`. The tree has already been moved, probably from different disks, possibly some directories using the root account, during which some directories may have been found to be duplicate, hence the different ownership (the btrfs snapshot on the probably source disk does not have root ownership there).

**Comment by xaizek** [comment]
Sorry to hear that some data might have been lost. However, I didn't get how it happened. Original file is removed only after copy is done and copy is not deleted on error. Destination is deleted on overwrite, but for that there need to be multiple conflicts per destination file, which shouldn't happen unless there is a bug related to that or multiple directories with the same name were copied simultaneously. And if original wasn't removed, what was lost?

> The process of moving ends prematurely.

What did you choose from `[r]etry/[i]gnore/[I]gnore for all/[a]bort` options when `Failed to unlink file` error was shown?

> In case this ends up being related: In the moved directory tree, some directories belong to root 

It probably is, although simpler way to get that error is to do `:chmod -w` on a directory. One strange thing I just noticed is that during merging conflict resolution for a file appears twice.

Also, how many directories participated in moves? Was it only one or multiple ones?

**Comment by aleksejrs** [comment]
>>  The process of moving ends prematurely.

> What did you choose from [r]etry/[i]gnore/[I]gnore for all/[a]bort options when Failed to unlink file error was shown?

[I]gnore for all

> Also, how many directories participated in moves? Was it only one or multiple ones?

I am not sure. I think multiple ones at least in some cases.

**Comment by aleksejrs** [comment]
I've just deleted a few files without moving them and got the error (the files disappeared). There is nothing strange about their ownership.

**Comment by aleksejrs** [comment]
It was more like one of the messages from the issue description. There was nothing like that in dmesg

Though syslog contains groups of messages like the following (not that GitHub says my comment was posted on 21:55, so that group of messages appears later), but apparently they are caused by a missing dependency, and may have happened because I was mounting a smartphone or an SD card.

`Jan 11 21:59:06 d udisksd[1538]: The function 'bd_crypto_luks_get_metadata_size' called, but not implemented!`

**Comment by aleksejrs** [comment]
I haven't seen this happen since then. I am not familiar with file name restrictions for btrfs like those mentioned for exFAT in #408.

**Comment by aleksejrs** [comment]
I guess I ignored the word "ignored" everywhere in the first message of #408. This must be a duplicate now.

**Comment by xaizek** [comment]
On the other hand, the error message was different. So I'm not all that sure. Anyway, let's consider it a duplicate for now and revise if something new will show up.

**Comment by xaizek** [comment]
Duplicate of #408
